plugins/woocommerce/includes/tracks/class-wc-tracks-event.php

```diff
@@ -92,6 +92,16 @@ public static function validate_and_sanitize( $event ) {
 			$_event->_ts = WC_Tracks_Client::build_timestamp();
 		}
 
+		if ( ! self::event_name_is_valid( $_event->_en ) ) {
+			return new WP_Error( 'invalid_event_name', __( 'A valid event name must be specified.', 'woocommerce' ) );
+		}
+
+		foreach ( array_keys( (array) $_event ) as $key ) {
+			if ( ! self::prop_name_is_valid( $key ) ) {
```

Comment:
I did do some general things around WC, but wasn't able to trigger the error that might use the above props.

Reply:
Yes, I already saw that method, but since it added more validations than what we needed here I preferred to add mine in `validate_and_sanitize` directly.

Reply:
Yeah I don't mind that you added that logic to the `validate_and_sanitize`, I was just wondering if there are any other exceptions to that rule that we could be missing.

Reply:
I couldn't find any 🤔